### DISCUSSION TITLE
Routers suggestion - accept functions for options

### DISCUSSION THREAD

**Discussion by frnora** [clientIO/joint#2241] [Ideas]
Hi,

I am trying to figure out the best way to have the manhattan router always come and go to the preferred side.

As title suggests, I think it would be nice to be able to provide functions for some of the router options. Namely:
1. manhattan: `endDirections` and `startDirections`
2. metro: `endDirections` and `startDirections`
3. rightAngle: `sourceDirection` and `targetDirection`

Currently the default router needs to be replaced or its `args` updated which (most likely) means it needs to be then restored to the default value later on.

Thanks for reading.

**Comment by kumilingus** [comment]
You can wrap the `manhattan` router in a function and determine the arguments e.g. `startDirections` inside this function.
See https://jsfiddle.net/kumilingus/w0rkposx/.

**Reply by frnora** [reply]
Great solution, thanks for it!
